src/main/java/com/ldtteam/structurize/items/ItemScanTool.java

```diff
@@ -241,12 +233,479 @@ public boolean canAttackBlock(final BlockState state, final Level worldIn, final
                 {
                     RenderingCache.queue("scan", new BoxPreviewData(((IBlueprintDataProviderBE) te).getInWorldCorners().getA(), ((IBlueprintDataProviderBE) te).getInWorldCorners().getB(), Optional.of(pos)));
                 }
-                itemstack.getOrCreateTag().put(NBT_START_POS, NbtUtils.writeBlockPos(start));
-                itemstack.getOrCreateTag().put(NBT_END_POS, NbtUtils.writeBlockPos(end));
+                setBounds(itemstack, start, end);
             }
         }
 
-        itemstack.getOrCreateTag().put(NBT_ANCHOR_POS, NbtUtils.writeBlockPos(pos));
+        setAnchorPos(itemstack, pos);
         return false;
     }
+
+    @Override
+    public void appendHoverText(@NotNull ItemStack stack,
+                                @Nullable Level world,
+                                @NotNull List<Component> tooltip,
+                                @NotNull TooltipFlag flags)
+    {
+        super.appendHoverText(stack, world, tooltip, flags);
+
+        if (stack.hasTag())
+        {
+            tooltip.add(getCurrentSlotDescription(stack));
+        }
+    }
+
+    @Override
+    public Component getHighlightTip(@NotNull final ItemStack stack, @NotNull final Component displayName)
+    {
+        return Component.empty()
```

Comment:
in previously versions using component.empty() equivalent for creating new components wasn't safe, does this method creates new instance of component or returns static singleton?

Reply:
The `ComponentContents` is a singleton, the outer `MutableComponent` is not.  So I'm reasonably sure this is safe.  Certainly appears to work fine.  Definitely something to bear in mind when backporting, though, that code is different in 1.18 so it might be unsafe there.

Reply:
I've found some vanilla code that uses the exact same pattern to create new mutable components, so it's definitely safe.  Will still keep it in mind for the backport though.